src/main/java/calculator/util/Operation.java

```diff
@@ -0,0 +1,53 @@
+package calculator.util;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public enum Operation {
+    PLUS("+") {
+        public double apply(double x, double y) {
+            return x + y;
+        }
+    },
+    MINUS("-") {
+        public double apply(double x, double y) {
+            return x - y;
+        }
+    },
+    TIMES("*") {
+        public double apply(double x, double y) {
+            return x * y;
+        }
+    },
+    DIVDE("/") {
+        public double apply(double x, double y) {
+            return x / y;
+        }
+    };
+
+    public abstract double apply(double x, double y);
+
+    private final String symbol;
+
+    Operation(String symbol) {
+        this.symbol = symbol;
+    }
+
+    @Override
+    public String toString() {
+        return symbol;
+    }
+
+    private static final Map<String, Operation> stringToEnum =
+        Stream.of(values()).collect(Collectors.
+            toMap(Object::toString, e -> e));
+
+    /**
+     fromString은 toString이 반환하는 문자열을 해당 열거 타입 상수로 변환해주는 기능
+     */
+    public static Optional<Operation> fromString(String symbol) {
+        return Optional.ofNullable(stringToEnum.get(symbol));
+    }
```

Comment:
```suggestion
  public static Operation from(String symbol) {
        return Arrays.stream(values())
                .filter(operation -> symbol.equals(operation.symbol))
                .findFirst().orElseThrow(IllegalArgumentException::new);
    }
```

이렇게도 표현이 가능합니다 : ) 
    
또한 하나의 매개변수를 받아서 인스턴스를 생성할 때 주로 `from`이라는 네이밍을 사용합니다. :)
네이밍에 관해서는 아래 링크 참고해주세요!
https://multifrontgarden.tistory.com/255

Reply:
스트림 사용에 미숙했는데, 저렇게 사용할 수도 있군요!... stream API 에 대해 공부해야겠다는 생각이 드는군요
조언 주신대로 바꾸었고, 바뀐 코드에 맞춰 기존 Calculator에서 Optional null처리 부분을 삭제하였습니다!

* 그리고 클래스와 메소드에 대한 전반적인 네이밍을 다시 하여 적용하였습니다 : )